stereo_points/stereo_points.py

```diff
@@ -203,96 +201,79 @@ def onControlPointSelectedChanged(self, newNode):
 
         if slicer.mrmlScene.GetNodesByName(coordTableName).GetNumberOfItems() == 0:
             # print('create new table')
-            coordTable = slicer.vtkMRMLTableNode()
-            coordTable.SetName(coordTableName)
-            coordTable.SetLocked(True)
+            self.coordTableNode = slicer.vtkMRMLTableNode()
+            self.coordTableNode.SetName(coordTableName)
+            self.coordTableNode.SetLocked(True)
             for col in ['Marker', 'x', 'y', 'z', 'r', 'a', 'd', 'X', 'Y', 'Z', 'R', 'A', 'S', 'i', 'j', 'k']:
-                c = coordTable.AddColumn()
+                c = self.coordTableNode.AddColumn()
                 c.SetName(col)
 
-            slicer.mrmlScene.AddNode(coordTable)
-            # keep a map of the fiducialNode <-> TableNode couples
-            self.fiducialNodesToCoordTables_id_tuples.append((newNode.GetID(), coordTable.GetID()))
+            slicer.mrmlScene.AddNode(self.coordTableNode)
+            # refererence the line from the table so we can get back to it
+            self.coordTableNode.AddNodeReferenceID("vtkMRMLMarkupsLineNode", newNode.GetID())
+            newNode.AddNodeReferenceID("vtkMRMLTableNode", self.coordTableNode.GetID())
             # add an observer for both nodes, whenever one is renamed, the other one is renamed as well
-            coordTable.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
+            self.coordTableNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
             newNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onControlPointNodeModified)
         else:
             # print('found it !')
-            coordTable = slicer.mrmlScene.GetNodesByName(coordTableName).GetItemAsObject(0)
+            self.coordTableNode = slicer.mrmlScene.GetNodesByName(coordTableName).GetItemAsObject(0)
```

Comment:
It is not nice that we take over whatever is the first table in the scene. If we use node references then that should be usable to tell if we have created a table node for this case already. If for some reason we cannot use a reference then at least we should add a node attribute to the tables that this module creates (e.g., `self.coordTableNode.SetAttribute("Stereotaxia", "1")` and we can take the first table node that has that attribute set, not the first node in the scene (that might have been created a different module and may be completely unrelated).